client/serviceinterface_update.go

```diff
@@ -329,7 +329,11 @@ func removeServiceInterfaceTarget(serviceName string, targetName string, deleteI
 			modified := false
 			targets := []types.ServiceInterfaceTarget{}
 			for _, t := range service.Targets {
-				if (t.Name == targetName || (t.Name == "" && targetName == serviceName)) && (t.Namespace == "" || t.Namespace == namespace) {
+				targetName, namespace = kube.GetServiceName(targetName, namespace)
```

Comment:
I think the logic here should be that we remove if targetName == t.Service || ((t.Service == "" && targetName == t.Name) && (namespace == "" || namespace == t.Namespace)). I.e. we don't need to care about the namespace at all if the Service on the target is specified, nor about any interpretation of whether the service is qualified or not.

Reply:
Logic has been updated and feels clearer now.